[DirectX][objdump] Add support for printing signatures

llvm-beanz opened the pull request:
This adds support for printing the signature sections as part of the `-p` flag for printing private headers.

The formatting aims to roughly match the formatting used by DXC's `/dumpbin` flag.

The original version's printed output left some trailing whitespace on lines, which caused the tests to fail with the strict whitespace matching.

Re-lands #152531.
Resolves #152380.

llvm-beanz committed:
[DirectX][objdump] Add support for printing signatures (#152531)

This adds support for printing the signature sections as part of the
`-p` flag for printing private headers.

The formatting aims to roughly match the formatting used by DXC's
`/dumpbin` flag.

Resolves #152380.

../llvm/test/tools/llvm-objdump/DXContainer/input-output-signatures.yaml

jh7370 commented:
No idea why, but the pre-merge checks are failing on both Windows and Linux due to the new test.

llvm-beanz committed:
Handle 0 safely

If the input number is 0, the result of log10 is unrepresentable as an
integer, which is UB.

llvm-beanz commented:
> No idea why, but the pre-merge checks are failing on both Windows and Linux due to the new test.

To close the loop here, the issue was mishandling 0 when calculating the number of digits needed needed to set the column widths. The failure was an UBSan issue, and is corrected with my latest push.

jh7370 reviewed (approved):
LGTM, thanks!
